### PR TITLE
Test cache affinity

### PR DESCRIPTION
**Pull request by aarshkshah1992**
Supersedes https://github.com/filecoin-saturn/caboose/pull/162.

### CODE REVIEW

File: pool_dynamics_test.go [hunk 1]
```diff
@@ -265,38 +267,32 @@ func TestPoolAffinity(t *testing.T) {
...
+			for _, n := range badNodes {
+				n := n
+				if n.URL == nodes[0].URL {
+					rerouteCount++
+				}
+			}
 		}
 
+		// no more than 5 cids from the cid list of 20 should get re-routed (25%)
+		assert.LessOrEqual(t, rerouteCount, 4)
```
Comment (willscott) [inline pool_dynamics_test.go]: should this be 5, rather than 4?

File: pool_dynamics_test.go [hunk 2]
```diff
@@ -265,38 +267,32 @@ func TestPoolAffinity(t *testing.T) {
...
 			}
 			nodes, _ := ch.CabooseActiveNodes.GetNodes(aff, ch.Config.MaxRetrievalAttempts)
 
-			// We expect that the candidate nodes are part of the "good nodes" list.
-			assert.Contains(t, goodNodes, nodes[0])
+			for _, n := range badNodes {
+				n := n
+				if n.URL == nodes[0].URL {
+					rerouteCount++
+				}
```
Comment (willscott) [inline pool_dynamics_test.go:290]: should 'reroute' be incremented when this doesn't match?

Comment (AmeanAsad) [reply pool_dynamics_test.go:290]: I believe reroute is meant to keep track of how often caboose returns a new joining node as the candidate node to fetch a cid in which case caboose is "rerouting" where it fetches that cid from. In that case, the increment here is good imo.

Comment (aarshkshah1992) [reply pool_dynamics_test.go:290]: Yeah what Amean said.

Comment (aarshkshah1992) [reply pool_dynamics_test.go:290]: We're iterating over bad nodes here.

**Comment by AmeanAsad** [comment]
Hey @aarshkshah1992 I think we should change notation of "bad" and "good" nodes in the affinity tests to maybe just "baseNodes" and "newNodes". I know I started with that notation in the test but I think its better to change it to make the test a bit more clear.

**Comment by aarshkshah1992** [comment]
@AmeanAsad Done.

File: pool_dynamics_test.go (hunk 1 above)
Comment (aarshkshah1992) [reply pool_dynamics_test.go]: Done.

**Review by AmeanAsad — APPROVED** [review]
Looks good!

**Pull request merged**